Calculate/Program.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Numerics;
+
+namespace Calculate;
+
+public class Program
+{
+    public Action<string?> WriteLine { get; init; }
+    public Func<string?> ReadLine { get; init; }
+    public Program()
+    {
+        WriteLine = Console.WriteLine;
+        ReadLine = Console.ReadLine;
+    }
+
+    public Program(Action<string?> writeLine, Func<string?> readLine)
+    {
+        WriteLine = writeLine ?? throw new ArgumentNullException(nameof(writeLine));
+        ReadLine = readLine ?? throw new ArgumentNullException(nameof(readLine));
+    }
+
+    public int Run<T>(Calculator<T> calculator) where T : INumber<T>
+    {
+        ArgumentNullException.ThrowIfNull(calculator);
+
+        while (true)
```

Comment:
Consider that having an infinite loop creates a certain kind of interactive experience like a [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop) vs if you did not have the loop the app would exit on each invocation and could be used like in an automated tool call fashion.